### PR TITLE
Speed up missing._get_interpolator

### PR DESCRIPTION
**Pull request by Illviljan**
Importing scipy.interpolate is slow and should only be done when necessary. Test case from 200ms to 6ms.

- [x] Related to #4739
- [x] Passes `isort . && black . && mypy . && flake8`

<sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior per commit by adding a `[test-upstream]` tag to the first line of the commit message.</sub>

### CODE REVIEW

**Review by mathause — APPROVED** [review]

File: xarray/core/missing.py [hunk 1]
```diff
@@ -437,6 +437,16 @@ def bfill(arr, dim=None, limit=None):
...
 
 
+def _import_interpolant(interpolant, method):
+    """Import interpolant from scipy.interpolate."""
+    try:
+        from scipy import interpolate
+
+        return getattr(interpolate, interpolant)
+    except ImportError:
+        raise ImportError(f"Interpolation with method {method} requires scipy")
```
Comment (keewis) [inline xarray/core/missing.py:447]: can we do something like `raise ImportError(...) from e` here? That way we don't drop the additional information the original `ImportError` might have, which is useful when the `ImportError` is not a `ModuleNotFoundError`.

**Review by dcherian — APPROVED** [review]

File: xarray/core/missing.py (hunk 1 above)
Comment (Illviljan) [reply xarray/core/missing.py]: Done.

Comment (keewis) [reply xarray/core/missing.py]: great, thanks.

**Review by keewis — APPROVED** [review]

**Comment by dcherian** [comment]
Thanks @Illviljan

**Pull request merged**